### PR TITLE
initial version of training material for small ncRNA clustering using blockclust

### PR DESCRIPTION
**Pull request by pavanvidem**

### CODE REVIEW

File: topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+>
+{: .hands_on}
+
+The `BlockClust: BED of predicted clusters` file is the result of Markov cluster algorithm. Each entry in this file represents a blockgroup. The 4th column contains is in the form of *annotation:blockgroup_id:cluster_id*. The *cluster_id* represents which cluster the blockgroup belongs to.
+
+The `BlockClust: Model based predictions BED` file is the result of classification. Each entry in this file represents a blockgroup. The 4th column contains the annotation of the ncRNA which it overlaps. For the blockgroups which are not overlapped with any known ncRNAs, there is a prefix 'predicted_' is added indicating that it is predicted by the **BlockClust** classification models.
+
+Next, we will visualize the read profiles (from the BAM file) and the predictions (BED from **BlockClust**) together.
+
+> ### {% icon hands_on %} Hands-on: visualization
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]: Additionally no newline should be between `>` and a `{:...}` tag

File: topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+We start with a BAM file as an input for this tutorial. If you want to use your own sequencing data, please clip the adapters and map to the reference genome. For this tutorial purpose, we already clipped the adapters and mapped the reads using **segemehl**.
+
+First, we convert the BAM file into BED file. This is not a plain file conversion. Hence we cannot use any BAM to BED conversion tools. We use BlockClust tool in the pre-processing mode for this purpose. The resulting BED file contains tags (a tag is a unique read sequence in a deep-sequencing library) and their normalized expression (column 5), i.e. the ratio of the read count per tag to the number of mappings on the reference genome.
+{: .hands_on}
+## Get data
+
+> ### {% icon hands_on %} Hands-on: Data upload
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo]() or from the shared data library
```
Comment (erasche) [inline topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]: Missing link I think

File: topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+{: .hands_on}
+
+## Group reads into blocks and blockgroups using **blockbuster**
+
+> ### {% icon hands_on %} Hands-on: blockbuster
+>
+> 1. **blockbuster** {% icon tool %} with the following parameters:
+>    - {% icon param-file %} *"BED file containing read expressions"*: `outfile` (output of **Sort** {% icon tool %})
+>
+>    ***TODO***: *Check parameter descriptions*
```
Comment (erasche) [inline topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]: Will you fill these in with anything? Or just remove them?

File: topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+> >
+> > Generally, the processed RNA fragments from the small RNAs are about 18-30nt long. If the sequenced read length is longer than the fragment (which is usually the case), the sequencer reads into the 3' adapter.
+> >
+> {: .solution}
+>
+{: .question}
+
+
+## Sort BAM file
+> Before continue to BAM to BED conversion we need to sort the alignments in the input BAM file by their positions.
```
[the remark this answers is not in the document]
Comment (pavanvidem) [reply topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]: @erasche thanks a lot for the suggestions! Hopefully, covered everything. The workflow is already unde https://github.com/usegalaxy-eu/workflow-testing/tree/master/blockclust. Should I move it to https://github.com/usegalaxy-eu/workflow-testing/tree/master/training/transcriptomics/ ?

[the remark this answers is not in the document]
Comment (erasche) [reply topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]: Ah, ok, I missed it. I'll fix that, no worries.

File: topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+# Clustering with **BlockClust**
+Here we use **BlockClust** in clustering mode. All you need here is the output of the **blockbuster**.
+
+Apart from clustering, **BlockClust** has built-in class specific discriminative models for C/D box snoRNA, H/ACA box snoRNA, miRNA, rRNA, snRNA, tRNA and Y_RNA. So it can also be used to predict if a read profile might belong to one of the known ncRNA class.
+> ### {% icon hands_on %} Hands-on: clustering
+>
+> 1. **BlockClust** {% icon tool %} with the following parameters:
+>    - *"Select mode of operation"*: `Clustering and classification`
+>        - {% icon param-file %} *"Input blockgroups file"*: output of **blockbuster** {% icon tool %}
+>        - *"Select reference genome"*: `Humann (hg19)`
```
Comment (erasche) [inline topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]:
```suggestion
>        - *"Select reference genome"*: `Human (hg19)`
```

File: topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+> ### {% icon hands_on %} Hands-on: visualization
+>
+> 1. Install [IGV](https://software.broadinstitute.org/software/igv/download) (if not already installed)
+> 2. Start IGV locally
+> 3. Expand the {% icon param-file %} `GSM769512.bam` file
+> 4. Click on the `local` in `display with IGV` to load the reads into the IGV browser
+> 5. Expand the {% icon param-file %} `BlockClust: Model based predictions BED` file
+> 6. Click on the `local` in `display with IGV` to load the reads into the IGV browser
+> 7. Go to the location `chr4:90653059-90653141` on IGV.
```
Comment (erasche) [inline topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.md]:
```suggestion
> 7. Go to the location `chr4:90653059-90653141` in IGV.
```

**Comment by erasche** [comment]
Looks much better! Thanks @pavanvidem

**Comment by pavanvidem** [comment]
thanks a lot @erasche ! There is already a workflow here https://github.com/usegalaxy-eu/workflow-testing/tree/master/blockclust. should I move it under https://github.com/usegalaxy-eu/workflow-testing/tree/master/training/transcriptomics ?

**Comment by erasche** [comment]
@pavanvidem I'll take care of that, no worries.

**Pull request merged**